docs/tasks/configure-pod-container/environment-variable-expose-pod-information.md

```diff
@@ -0,0 +1,164 @@
+---
+title: Exposing Pod Information Through Environment Variables
+---
+
+{% capture overview %}
+
+This page shows how a Pod can use environment variables to expose information
+about itself to Containers running in the Pod. Environment variables can expose
+Pod fields and Container fields.
+
+There are two ways to expose Pod and Container fields to a running Container:
+environment variables and
+[DownwardAPIVolumeFiles](/docs/resources-reference/v1.5/#downwardapivolumefile-v1).
+Together, these two ways of exposing Pod and Container fields are called the
+*Downward API*.
+
+{% endcapture %}
+
+
+{% capture prerequisites %}
+
+{% include task-tutorial-prereqs.md %}
+
+{% endcapture %}
+
+
+{% capture steps %}
+
+## Using Pod fields as values for environment variables
+
+In this exercise, you create a Pod that has one Container. Here is the
+configuration file for the Pod:
+
+{% include code.html language="yaml" file="dapi-envars-pod.yaml" ghlink="/docs/tasks/configure-pod-container/dapi-envars-pod.yaml" %}
+
+In the configuration file, you can see five environment variables. The `env`
+field is an array of
+[EnvVars](/docs/resources-reference/v1.5/#envvar-v1).
+The first element in the array specifies that the `MY_NODE_NAME` environment
+variable gets its value from the Pod's `spec.nodeName` field. Similarly, the
+other environment variables get their names from Pod fields.
+
+**Note**: The fields in this example are Pod fields. They are not fields of the
+Container in the Pod.
+
+Create the Pod:
+
+```shell
+kubectl create -f http://k8s.io/docs/tasks/configure-pod-container/dapi-envars-pod.yaml
+```
+
+Verify that the Container in the Pod is running:
+
+```
+kubectl get pods
+```
+
+View the Container's logs:
+
+```
+kubectl logs dapi-envars-fieldref
+```
+
+The output shows the values of selected environment variables:
+
+```
+minikube
+dapi-envars-fieldref
+default
+172.17.0.4
+default
+```
+
+To see why these values are in the log, look at the `command` and `args` fields
+in the configuration file. When the Container starts, it writes the values of
+five environment variables to stdout. It repeats this every ten seconds.
+
+Next, get a shell into the Container that is running in your Pod:
+
+```
+kubectl exec -it dapi-envars-fieldref -- sh
+```
+
+In your shell, view the environment variables:
+
+```
+/ # printenv
```

Comment:
In other examples I've seen, the shell prompt is "/#" without the space. Which one is correct? Or does it not matter?